proto/osmosis/poolmanager/v1beta1/query.proto

```diff
@@ -248,4 +256,24 @@ message TotalLiquidityResponse {
     (gogoproto.moretags) = "yaml:\"liquidity\"",
     (gogoproto.nullable) = false
   ];
-}
\ No newline at end of file
+}
+
+//=============================== EstimateTradeBasedOnPriceImpact
+message EstimateTradeBasedOnPriceImpactRequest {
+  cosmos.base.v1beta1.Coin from_coin = 1 [ (gogoproto.nullable) = false ];
+  string to_coin_denom = 2;
+  uint64 pool_id = 3;
+  string max_price_impact = 4 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  string twap_price = 5 [
```

Comment:
add documentation for this and maybe change the name

Reply:
We can use something like `external_price` or `reference_price`

Reply:
I ll change to `external_price`